mlflow/store/_unity_catalog/registry/rest_store.py

```diff
@@ -0,0 +1,251 @@
+import logging
+
+from mlflow.protos.databricks_uc_registry_messages_pb2 import (
+    CreateRegisteredModelRequest,
+    CreateRegisteredModelResponse,
+    UpdateRegisteredModelRequest,
+    UpdateRegisteredModelResponse,
+    DeleteRegisteredModelRequest,
+    DeleteRegisteredModelResponse,
+    CreateModelVersionRequest,
+    CreateModelVersionResponse,
+    FinalizeModelVersionRequest,
+    FinalizeModelVersionResponse,
+    UpdateModelVersionRequest,
+    UpdateModelVersionResponse,
+    DeleteModelVersionRequest,
+    DeleteModelVersionResponse,
+    GetModelVersionDownloadUriRequest,
+    GetModelVersionDownloadUriResponse,
+    SearchModelVersionsRequest,
+    SearchModelVersionsResponse,
+    GetRegisteredModelRequest,
+    GetRegisteredModelResponse,
+    GetModelVersionRequest,
+    GetModelVersionResponse,
+    SearchRegisteredModelsRequest,
+    SearchRegisteredModelsResponse,
+    GenerateTemporaryModelVersionCredentialsRequest,
+    GenerateTemporaryModelVersionCredentialsResponse,
+)
+from mlflow.exceptions import MlflowException
+from mlflow.protos.databricks_uc_registry_service_pb2 import UcModelRegistryService
+from mlflow.store.entities.paged_list import PagedList
+from mlflow.utils.proto_json_utils import message_to_json
+from mlflow.utils.rest_utils import (
+    extract_api_info_for_service,
+    extract_all_api_info_for_service,
+    _REST_API_PATH_PREFIX,
+)
+from mlflow.store.model_registry.rest_store import BaseRestStore
+from mlflow.store._unity_catalog.registry.utils import (
+    registered_model_from_uc_proto,
+)
+
+_METHOD_TO_INFO = extract_api_info_for_service(UcModelRegistryService, _REST_API_PATH_PREFIX)
+_METHOD_TO_ALL_INFO = extract_all_api_info_for_service(
+    UcModelRegistryService, _REST_API_PATH_PREFIX
+)
+
+_logger = logging.getLogger(__name__)
+
+
+def _require_arg_unspecified(arg_name, arg_value, default_value=None, message=None):
+    if arg_value != default_value:
+        _raise_unsupported_arg(arg_name, message)
+
+
+def _raise_unsupported_arg(arg_name, message=None):
+    messages = [
+        f"Argument {arg_name} is unsupported for models in the Unity Catalog.",
+    ]
+    if message is not None:
+        messages.append(message)
+    messages.append("See the user guide for more information")
```

Comment:
Given that we'll be in private preview, it's not possible to link to a specific doc page explaining which args/APIs are unsupported and how to handle that, so we just point customers to the user guide for now. At public preview time we can improve this